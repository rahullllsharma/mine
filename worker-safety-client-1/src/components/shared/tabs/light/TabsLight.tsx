import type { ItemClasses, TabsOption, TabsProps } from "../Tabs";
import Tabs from "../Tabs";

export default function TabsLight({
  options,
  defaultIndex,
  onSelect,
}: Omit<TabsProps, "itemClasses" | "options"> & {
  options: Pick<TabsOption, "id" | "name">[] | string[];
}): JSX.Element {
  const itemClasses: ItemClasses = {
    default:
      "text-neutral-shade-75 font-semibold active:bg-neutral-shade-3 border-b-4",
    selected: "border-brand-urbint-40 border-solid z-10",
    unselected: "opacity-75 border-transparent",
  };

  return (
    <>
      <Tabs
        itemClasses={itemClasses}
        options={options}
        defaultIndex={defaultIndex}
        onSelect={onSelect}
      />
      <hr
        className="relative bg-neutral-shade-18 z-1 border"
        style={{
          bottom: "3px",
        }}
      />
    </>
  );
}
