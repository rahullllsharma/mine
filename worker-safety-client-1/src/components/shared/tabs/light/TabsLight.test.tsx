import { render } from "@testing-library/react";
import TabsLight from "./TabsLight";

const options = ["Active", "Pending", "Completed"];

describe(TabsLight.name, () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(
      <TabsLight options={options} onSelect={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
