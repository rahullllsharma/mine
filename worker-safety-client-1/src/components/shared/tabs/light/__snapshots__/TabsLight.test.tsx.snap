// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsLight should match the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-orientation="horizontal"
    class="flex"
    role="tablist"
  >
    <button
      aria-selected="true"
      class="min-w-0 z-10 outline-none"
      id="headlessui-tabs-tab-1"
      role="tab"
      tabindex="0"
      type="button"
    >
      <div
        class="flex items-center px-2.5 py-3 text-neutral-shade-75 font-semibold active:bg-neutral-shade-3 border-b-4 border-brand-urbint-40 border-solid z-10"
      >
        <span
          class="text-base mx-1 truncate"
        >
          Active
        </span>
      </div>
    </button>
    <button
      aria-selected="false"
      class="min-w-0 z-10 outline-none"
      id="headlessui-tabs-tab-2"
      role="tab"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex items-center px-2.5 py-3 text-neutral-shade-75 font-semibold active:bg-neutral-shade-3 border-b-4 opacity-75 border-transparent"
      >
        <span
          class="text-base mx-1 truncate"
        >
          Pending
        </span>
      </div>
    </button>
    <button
      aria-selected="false"
      class="min-w-0 z-10 outline-none"
      id="headlessui-tabs-tab-3"
      role="tab"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex items-center px-2.5 py-3 text-neutral-shade-75 font-semibold active:bg-neutral-shade-3 border-b-4 opacity-75 border-transparent"
      >
        <span
          class="text-base mx-1 truncate"
        >
          Completed
        </span>
      </div>
    </button>
  </div>
  <hr
    class="relative bg-neutral-shade-18 z-1 border"
    style="bottom: 3px;"
  />
</DocumentFragment>
`;
